### PR TITLE
Fix linked non-cluster-default blob storage not working for HDI cluster issue

### PR DESCRIPTION
**Pull request by t-rufang**

### CODE REVIEW

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmissionParameter.java [hunk 1]
```diff
@@ -327,6 +327,14 @@ private Integer parseIntegerSafety(@Nullable Object maybeInteger) {
         return messageList;
     }
 
+    @JsonIgnore
+    public Map<String, Object> updateJobConfig(String key, Object value) {
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmissionParameter.java:331]: I prefer to keep this file minimal changes. We can do it at calling side as below:
```
submitModel.submissionParameter.getJobConfig().put(SparkSubmissionParameter.Conf, ...)
```

**Review by wezhang — COMMENTED** [review]
Thanks Rui!

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/WasbUri.java [hunk 2]
```diff
@@ -90,7 +90,7 @@ public int hashCode() {
     }
 
     public String getHadoopBlobFsPropertyKey() {
-        return String.format("fs.azure.account.key.%s.blob.%s", getStorageAccount(), getEndpointSuffix());
+        return String.format("spark.hadoop.fs.azure.account.key.%s.blob.%s", getStorageAccount(), getEndpointSuffix());
```
[the remark this answers is not in the document]
Comment (t-rufang) [reply Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/WasbUri.java:93]: Good suggestion, will make the change in next iteration

**Review by wezhang — APPROVED** [review]
LGTM! Shipit!

**Pull request merged**